src/game/logic/object/weapon.cpp

```diff
@@ -101,10 +102,253 @@ void WeaponStore::Parse_Weapon_Template(INI *ini, void *, void *store, const voi
+}
+
+void Weapon::CRC_Snapshot(Xfer *xfer)
+{
+#ifdef GAME_DEBUG_STRUCTS
+    Utf8String str1;
+    Utf8String str2;
+    // todo
+#endif
+
+    Utf8String name = m_template->Get_Name();
+    xfer->xferAsciiString(&name);
+    xfer->xferUser(&m_wslot, sizeof(m_wslot));
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferUnsignedInt(&m_ammoInClip);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferUnsignedInt(&m_whenWeCanFireAgain);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferUnsignedInt(&m_whenPreAttackFinished);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferUnsignedInt(&m_whenLastReloadStarted);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferUnsignedInt(&m_lastFireFrame);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferObjectID(&m_projectileStreamID);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    ObjectID laser = OBJECT_UNK;
+    xfer->xferObjectID(&laser);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferInt(&m_maxShotCount);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferInt(&m_curBarrel);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferInt(&m_numShotsForCurBarrel);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    unsigned short scatter_count = static_cast<unsigned short>(m_scatterTargetIndexes.size());
+    xfer->xferUnsignedShort(&scatter_count);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    for (auto i = m_scatterTargetIndexes.begin(); i != m_scatterTargetIndexes.end(); i++) {
+        int index = *i;
+        xfer->xferInt(&index);
+
+#ifdef GAME_DEBUG_STRUCTS
+        // todo
+#endif
+    }
+
+    xfer->xferBool(&m_pitchLimited);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+
+    xfer->xferBool(&m_leechWeaponRangeActive);
+
+#ifdef GAME_DEBUG_STRUCTS
+    // todo
+#endif
+}
+
+void Weapon::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 3;
+    xfer->xferVersion(&version, 3);
+
+    if (version >= 2) {
+        Utf8String name;
+        name = m_template->Get_Name();
+        xfer->xferAsciiString(&name);
+
+        if (xfer->Get_Mode() == XFER_LOAD) {
+            m_template = g_theWeaponStore->Find_Weapon_Template(name);
+
+            if (m_template == nullptr) {
+                throw CODE_06;
+            }
+        }
+    }
+
+    xfer->xferUser(&m_wslot, sizeof(m_wslot));
+    xfer->xferUser(&m_status, sizeof(m_status));
+    xfer->xferUnsignedInt(&m_ammoInClip);
+    xfer->xferUnsignedInt(&m_whenWeCanFireAgain);
+    xfer->xferUnsignedInt(&m_whenPreAttackFinished);
+    xfer->xferUnsignedInt(&m_whenLastReloadStarted);
+    xfer->xferUnsignedInt(&m_lastFireFrame);
+
+    if (version < 3) {
+        m_suspendFXDelay = 0;
+    } else {
+        xfer->xferUnsignedInt(&m_suspendFXDelay);
+    }
+
+    xfer->xferObjectID(&m_projectileStreamID);
+    ObjectID laser = OBJECT_UNK;
+    xfer->xferObjectID(&laser);
```

Comment:
Is this laser variable used for anything?

Reply:
no, must be a leftover.